src/include/buffer.h

```diff
@@ -317,6 +318,36 @@ class CEPH_BUFFER_API buffer {
 
     };
 
+    class const_iterator : private iterator {
+    public:
+      const_iterator(const iterator& it)
+	: iterator(it) {}
+      const_iterator(const list *l, unsigned o=0)
+	: iterator(const_cast<list*>(l), o) {}
```

Comment:
So this is definitely cheating by using a non-const pointer internally. I'm not sure that's a good idea. :/

(You should definitely get other opinions! It just looks to me like asking for trouble.)